src/Provider/Provider.Web/Orchestrators/DashboardOrchestrator.cs

```diff
@@ -36,22 +49,49 @@ public async Task<DashboardViewModel> GetDashboardViewModelAsync(long ukprn)
                     v.NoOfApplications == 0),
                 NoOfVacanciesClosingSoon = vacancies.Count(v =>
                     v.ClosingDate <= _timeProvider.Today.AddDays(ClosingSoonDays) &&
-                    v.Status == VacancyStatus.Live)
+                    v.Status == VacancyStatus.Live),
+                TransferredVacanciesAlert = GetTransferredVacanciesAlertViewModel(dashboard.TransferredVacancies, userDetails.TransferredVacanciesAlertDismissedOn)
             };
             return vm;
         }
 
-        private async Task<List<VacancySummary>> GetVacanciesAsync(long ukprn)
+        public Task DismissAlert(VacancyUser user, AlertType alertType)
         {
-            var dashboard = await _client.GetDashboardAsync(ukprn);
+            return _client.UpdateUserAlertAsync(user.UserId, alertType, _timeProvider.Now);
+        }
+
+        private async Task<ProviderDashboard> GetDashboardAsync(long ukprn)
+        {
+            var dashboard = await _vacancyClient.GetDashboardAsync(ukprn);
 
             if (dashboard == null)
             {
-                await _client.GenerateDashboard(ukprn);
-                dashboard = await _client.GetDashboardAsync(ukprn);
+                await _vacancyClient.GenerateDashboard(ukprn);
+                dashboard = await _vacancyClient.GetDashboardAsync(ukprn);
             }
 
-            return dashboard?.Vacancies?.ToList() ?? new List<VacancySummary>();
-        }      
+            return dashboard;
+        }
+
+        private TransferredVacanciesAlertViewModel GetTransferredVacanciesAlertViewModel(IEnumerable<ProviderDashboardTransferredVacancy> transferredVacancies, DateTime? userLastDismissedDate)
+        {
+            if (transferredVacancies == null)
+                return null;
+
+            if (userLastDismissedDate.HasValue == false)
+                userLastDismissedDate = DateTime.MinValue;
+
+            var usersTransferredVacancies = transferredVacancies
+                .Where(t => t.TransferredDate > userLastDismissedDate)
+                .Select(t => t.LegalEntityName);
+            
+            if (usersTransferredVacancies.Any() == false)
+                return null;
+
+            return new TransferredVacanciesAlertViewModel
+            {
+                LegalEntityNames = usersTransferredVacancies.GroupBy(l => l).Select(l => l.Key)
```

Comment:
`usersTransferredVacancies.Distinct()` maybe as it already has only the LegalEntityNames projected I believe.

Reply:
👍